File: py_sol/1768-Merge Strings Alternately.py
def mergeAlternately(word1: str, word2: str) -> str:
        min_len = min(len(word1), len(word2))
        result = ""
        for i in range(min_len):
            result += word1[i] + word2[i]
        result+= word1[min_len:] + word2[min_len:]
        return result
    
result = mergeAlternately("ab","cdef")
print(result)

File: cpp_sol/1768-Merge Strings Alternately.cpp
#include <iostream>
using namespace std;
string mergeAlternately(string word1, string word2) {
    string word3 ="";
    const int minLen =min( word1.length(),word2.length());
    for (int i=0; i < minLen; ++i){
        word3 += word1[i];
        word3 += word2[i];
    }
    //deal with remaining strings
    return word3 + word1.substr(minLen)+word2.substr(minLen);
    }

int main(){
    string result = mergeAlternately("hello","world");
    cout << result << endl;
    return 0;
}